src/suites/cts/gpu_test.ts

```diff
@@ -157,12 +163,18 @@ export class GPUTest extends Fixture {
 
   // TODO: add an expectContents for textures, which logs data: uris on failure
 
-  expectContents(src: GPUBuffer, expected: ArrayBufferView): void {
-    const exp = new Uint8Array(expected.buffer, expected.byteOffset, expected.byteLength);
+  expectContents(src: GPUBuffer, expected: (
+    Int8Array | Int16Array | Int32Array |
+    Uint8Array | Uint16Array | Uint32Array |
+    Float32Array | Float64Array
+  )): void {
+    // @ts-ignore
+    const exp = new expected.constructor(expected.buffer, expected.byteOffset, expected.length);
```

Comment:
I found this really helpful so that failures are interpreted in terms of type of TypedArray you're using. That way I don't have to interpret multiple hex failures into float when looking at errors.

Reply:
I just mean I think this is now the same as:
```suggestion
    const exp = expected;
```